Enabling eslint rule for front: react-hooks/exhaustive-deps

lasryaric opened the pull request:
Enabling `react-hooks/exhaustive-deps` for front.

Here is why:
- enabling this rules gives us an eslint warning when we are missing dependencies in useEffect(), reducing the number of bugs in the UI. Especially bugs where the UI does not properly update to the new state because the useEffect() was not re-triggered on state change sit depends on.
- Forces us to wrap our inner functions in useCallback , which avoids re-rendering the UI (at least in the shadow DOM) too many times when nothing changed (but the reference to the inner function).

fontanierh commented on front/components/app/DatasetView.tsx:
```diff
@@ -144,9 +144,20 @@ export default function DatasetView({
 
     return valid;
-  };
+  }, [dataset?.name, datasetName, datasets]);
```
I'm not sure this is correct, because `datasets` is an array, so its reference could be the same even if the array was modified, no ? Is it required by `exhaustive-deps` that this must be a `useCallback` ?

fontanierh commented on front/components/app/DatasetView.tsx:
```diff
@@ -144,9 +144,20 @@ export default function DatasetView({
+        }, 10);
+      }
+    }
+  }, [datasetData, datasetInitializing, datasetKeys]);
```
same here, I don't think this is correct

lasryaric commented:
Found a bug while testing. Taking another look.

fontanierh reviewed (changes requested):
This seems more about adding a lot of `useCallback` and not really about turning on the `exhaustive-deps` rule ? I might be missing something, but I don't think arrays should be passed as hook dependencies

lasryaric commented:
@fontanierh your comments have been discussed IRL.
I have reverted all the code that is a bit old, well tested and too complicated to change today and added some lint ignore instructions.
PR is ready to merge if it's good on your side @fontanierh .
